content/en/docs/ops/configuration/extensibility/wasm-module-distribution/index.md

```diff
@@ -0,0 +1,143 @@
+  - /docs/ops/extensibility/distribute-remote-wasm-module
+keywords: [extensibility,Wasm,WebAssembly]
+owner: istio/wg-policies-and-telemetry-maintainers
+test: n/a
+---
+
+Istio provides the ability to [extend proxy functionality using WebAssembly (Wasm)](/blog/2020/wasm-announce/).
+One of the key advantages of Wasm extensibility is that extensions can be loaded dynamically at runtime.
+But first these extensions must be distributed to the proxy.
+Istio makes this possible by allowing the Istio agent to dynamically download Wasm modules.
+
+## Configure an HTTP Filter with a Remote Wasm Module
+
+Here we will walk through an example of adding a basic auth extension to our mesh. We will configure Istio to pull a [basic auth module](https://github.com/istio-ecosystem/wasm-extensions/tree/master/extensions/basic_auth) from a remote URI and load it with configuration to run the module on calls to the `/productpage` path.
+
+To configure a WebAssembly filter with a remote Wasm module, two `EnvoyFilter` resources will be installed: one injects the HTTP filter, and the other provides configuration for the filter to use the remote Wasm module.
+
+With the first `EnvoyFilter`, an HTTP filter will be injected into gateway proxies. It is configured to request the extension configuration named `istio.basic_auth` from `ads` (i.e. Aggregated Discovery Service), which is the same configuration source that Istiod uses to provide all other configuration resources. Within the configuration source, the initial fetch timeout is set to `0s`, which means that when the Envoy proxy processes a listener update with this filter, it will wait indefinitely for the first extension configuration update before accepting request with this listener.
+
+{{< text yaml >}}
+apiVersion: networking.istio.io/v1alpha3
+kind: EnvoyFilter
+metadata:
+ name: basic-auth
+ namespace: istio-system
+spec:
+ configPatches:
+ - applyTo: HTTP_FILTER
+   match:
+     context: GATEWAY
+     listener:
+       filterChain:
+         filter:
+           name: envoy.http_connection_manager
+   patch:
+     operation: INSERT_BEFORE
+     value:
+       name: istio.basic_auth
+       config_discovery:
+         config_source:
+           ads: {}
+           initial_fetch_timeout: 0s # wait indefinitely to prevent bad Wasm fetch
+         type_urls: [ "type.googleapis.com/envoy.extensions.filters.http.wasm.v3.Wasm"]
+{{< /text >}}
+
+The second `EnvoyFilter` provides configuration for the filter, which is an `EXTENSION_CONFIG` patch and will be distributed to the proxy as an Envoy [`Extension Configuration`](https://www.envoyproxy.io/docs/envoy/latest/configuration/overview/extension) (ECDS) resource.
+Once this update reaches the Istio agent, the agent will download the Wasm module and store it in the local file system.
+If the download fails, the agent will reject the `Extension Configuration` update to prevent invalid Wasm filter configuration from reaching the Envoy proxy.
```

Comment:
```suggestion
If the download fails, the agent will reject the ECDS update to prevent invalid Wasm filter configuration from reaching the Envoy proxy.
```